syntax="proto3";
package pms;

import "common_msg.proto";
import "sys_msg_type.proto";

option optimize_for = LITE_RUNTIME;

// this is for server module connecting connection
message ConnMsg{
    EConnTag           conn_tag = 1;
    ETransferModule    tr_module = 2; // the server module
    string             moduleid = 3;
    string             transferid = 4;
}

// this is for msg relaying between server
// this is used by TransferMsg's content field
message RelayMsg{
    EServerCmd         svr_cmds = 1;
    ETransferModule    tr_module = 2; // the server module
    EModuleType        cont_module = 3; // the msg content's module
    string             connector = 4;
    string             handle_cmd = 5;
    string             handle_mtype = 6;
    string             handle_data = 7;
    bytes              content = 8; // this is the application or module msg
    ToUser             touser = 9;
}

// this is for msg relaying between server
message TransferMsg{
    ETransferType type         = 1; // the type of transfer
    ETransferFlag flag         = 2; // the flag
    ETransferPriority priority = 3;
    bytes   content            = 4; // the message to transfer
}

// this is for system error msg, report to higher level
message ErrorMsg{
    ETransferModule emodule     = 1; // which module has error
    string          userid      = 2; // which user meet this error
    string          reason      = 3; // what reason for error
    int32           errcode     = 4; // err code
}
